src/clj/web/api.clj

```diff
@@ -81,11 +84,15 @@
 (defroutes tournament-routes
   (GET "/tournament-auth/:username" [] tournament/auth))
 
-(defroutes routes
+(defroutes private-routes
   (wrap-routes user-routes auth/wrap-authentication-required)
   (wrap-routes tournament-routes auth/wrap-tournament-auth-required)
-  (wrap-routes admin-routes auth/wrap-authorization-required)
-  public-routes)
+  (wrap-routes admin-routes auth/wrap-authorization-required))
+
+(defroutes routes
+  (wrap-routes private-routes wrap-anti-forgery)
+  (wrap-routes public-CSRF-routes wrap-anti-forgery)
+  public-routes wrap-anti-forgery)
```

Comment:
Should you move the `wrap-anti-forgery` down to the `app` definition?

Reply:
By default, the anti-forgery wrap will require the CSRF token on any endpoint that isnt a GET. If we apply it to all endpoints the login and register POST calls will require a CSRF... which you can only get by logging in.

What you are seeing here is a typo which I didnt pick up on in my testing because it is essentially ignored by the defroutes macro...

Reply:
cool, thanks